src/sessions/__tests__/VirtualPageLoadTimer.test.ts

```diff
@@ -0,0 +1,567 @@
+import { advanceTo } from 'jest-date-mock';
+import { PageManager } from '../PageManager';
+import {
+    DEFAULT_CONFIG,
+    mockFetch,
+    mockFetchWith500,
+    mockFetchWithError,
+    mockFetchWithErrorObject,
+    mockFetchWithErrorObjectAndStack
+} from '../../test-utils/test-utils';
+import { Config } from '../../orchestration/Orchestration';
+import { PERFORMANCE_NAVIGATION_EVENT_TYPE } from '../../plugins/utils/constant';
+import { PAGE_VIEW_EVENT_TYPE } from '../../plugins/utils/constant';
+import mock from 'xhr-mock';
+
+const record = jest.fn();
+
+declare const jsdom: any;
+
+Object.defineProperty(document, 'referrer', {
+    value: 'https://console.aws.amazon.com'
+});
+Object.defineProperty(document, 'title', { value: 'Amazon AWS Console' });
+global.fetch = mockFetch;
+
+describe('PageManager tests', () => {
+    let url;
+
+    beforeAll(() => {
+        url = window.location.toString();
+    });
+
+    beforeEach(() => {
+        advanceTo(0);
+        mock.setup();
+        record.mockClear();
+        jsdom.reconfigure({
+            url: url
+        });
+    });
+
+    afterEach(() => {
+        jest.resetModules();
+    });
+
+    test('when next page is created then items in requestBuffer should be added to ongoingRequests', async () => {
+        // Init
+        const config: Config = {
+            ...DEFAULT_CONFIG,
+            allowCookies: true
+        };
+        const pageManager: PageManager = new PageManager(config, record);
+        const helper = pageManager['virtualPageLoadTimer'];
+
+        // Create landing page
+        pageManager.recordPageView('/rum/home');
+
+        // Adding requests to requestBuffer
+        for (let i = 0; i < 3; i++) {
+            let xhr = new XMLHttpRequest();
+            helper['requestBuffer'].add(xhr);
+        }
+
+        // Create next page
+        pageManager.recordPageView('/console/home');
+
+        // Assert
+        expect(helper['ongoingRequests'].size).toEqual(3);
+        expect(helper['requestBuffer'].size).toEqual(0);
+        expect(helper['isPageLoaded']).toEqual(false);
```

Comment:
issue: Accessing private member variables is an anti-pattern (although *sometimes* necessary). This is because it is better to verify a property of the expected output rather than a property of the internal state.

In this case, we are verifying that the set of ongoing requests includes the requests from the requestBuffer. What we want is to verify that requests that started before the recordPageViewEvent will be included in the route change time.